### ISSUE TITLE
tour: Mutex isn't needed in example sync.Mutex

### ISSUE THREAD

**Issue by mattklein999** [golang/tour#35]
Context: https://tour.golang.org/concurrency/9

I removed the `Lock` and `Unlock` from both the `Inc` and `Value` methods and still keep getting 1000. I suspect that this may be some caching thing, or perhaps the program is only single threaded. Either way, it doesn't really drive home the importance of synchronization...

**Comment by adg** [comment]
Race conditions often present obscure symptoms, and I think it would be difficult to construct an example that reliably demonstrates a race condition in the playground sandbox (the code execution environment used by the tour).

Perhaps we should add some more words to the page to explain this.

**Comment by mattklein999** [comment]
fwiw, not being familiar with Go, I was looking at the code and thinking to myself that it was spinning up 1000 "threads," each of which did a read/write+1, and in my experience in C# this would have almost always manifested in a result significantly less than 1000.

I know it's an intro, but maybe just a little more depth on what a go routine is under the hood may have helped adjust my mental model of concurrency to be more in line with the go universe.

**Comment by adg** [comment]
> fwiw, not being familiar with Go, I was looking at the code and thinking to myself that it was spinning up 1000 "threads," each of which did a read/write+1, and in my experience in C# this would have almost always manifested in a result significantly less than 1000.

Yep, that's what it's doing, but they're being scheduled over a single thread (a constraint of the playground sandbox).

We should definitely add to the wording here. I also feel that this example is overly complicated.

**Comment by crisman** [comment]
Works now in playground, Mutex is needed.

In the current (early 2023 & go 1.20) playground commenting out `Lock()` and `Unlock()` in `Inc()` frequently generates a `fatal error: concurrent map writes` and rarely a number less than 1000.

**Comment by ALTree** [comment]
Thanks, we can close then.
